### PR TITLE
Feature 2.0 folder

### PR DESCRIPTION
**Pull request by ANioding**
Submitting code for folder feature.

### CODE REVIEW

File: src/main/java/com/smartsheet/api/FolderResources.java [hunk 1]
```diff
@@ -48,7 +49,7 @@
 	 * @throws ServiceUnavailableException if the REST API service is not available (possibly due to rate limiting)
 	 * @throws SmartsheetException if there is any other error during the operation
 	 */
-	public Folder getFolder(long folderId) throws SmartsheetException;
+	public Folder getFolder(long folderId, EnumSet<FolderInclusion> includes) throws SmartsheetException;
```
Comment (brettbatie) [inline src/main/java/com/smartsheet/api/FolderResources.java]: Use SourceInclusion which is available in the release-2.0 branch.

File: src/main/java/com/smartsheet/api/internal/FolderResourcesImpl.java [hunk 2]
```diff
@@ -64,8 +67,16 @@ public FolderResourcesImpl(SmartsheetImpl smartsheet) {
 	 * rather than returning null)
 	 * @throws SmartsheetException the smartsheet exception
 	 */
-	public Folder getFolder(long folderId) throws SmartsheetException {
-		return this.getResource("folder/" + folderId, Folder.class);
+	public Folder getFolder(long folderId, EnumSet<FolderInclusion> includes) throws SmartsheetException {
```
Comment (brettbatie) [inline src/main/java/com/smartsheet/api/internal/FolderResourcesImpl.java]: Remove checking for null on the includes as Ahmad is changing the generateUrl() to drop null parameters. The code can look like the following.

```
HashMap<String, String> parameters = new HashMap<String, String>();
parameters.put("include", QueryUtil.generateCommaSeparatedList(includes));
return this.getResource(QueryUtil.generateUrl(null, parameters), Folder.class);
```

File: src/main/java/com/smartsheet/api/FolderResources.java (hunk 1 above)
Comment (ANioding) [inline src/main/java/com/smartsheet/api/FolderResources.java]: Hey Brett, I have corrected and used sourceInclusion. Deleted FolderInclusion class from project.

File: src/main/java/com/smartsheet/api/internal/FolderResourcesImpl.java (hunk 2 above)
Comment (ANioding) [inline src/main/java/com/smartsheet/api/internal/FolderResourcesImpl.java]: Removed the check for nulls on includes parameter.

**Pull request merged**